File: py/000/ListNode.py

class ListNode(object):
    def __init__(self, x):
        self.val = x
        self.next = None

    @classmethod
    def getListNode(cls, *ints):
        re = ListNode(ints[0])
        prev = re
        for i in range(1, len(ints)):
            prev.next = ListNode(ints[i])
            prev = prev.next
        return re

    @classmethod
    def printListNode(cls, *nodes):
        bytes = []
        for ln in nodes:
            while ln != None:
                bytes.append(str(ln.val) + '\t')
                ln = ln.next
            bytes.append('\n')
        print(''.join(bytes))

ListNode.printListNode(ListNode.getListNode(1,5,6,2), ListNode.getListNode(7,8,6,2))

File: js/000/ListNode.js

function ListNode(val) {
    this.val = val;
    this.next = null;
}

// int... is -> ListNode
ListNode.getListNode = function() {
    let re = new ListNode(arguments[0]), prev = re;
    for (let i = 1; i < arguments.length; i++)
        prev = prev.next = new ListNode(arguments[i]);
    return re;
}

// ListNode... lns
ListNode.printListNode = function() {
    let bytes = [];
    for (let ln of arguments) {
        while (ln != null) {
            bytes.push(ln.val + '\t');
            ln = ln.next;
        }
        bytes.push('\n');
    }
    console.log(bytes.join(''));
}

ListNode.printListNode(ListNode.getListNode(1,5,6,2), ListNode.getListNode(7,8,6,2));
